doc/tutorial/mutli_arch.hpp

```diff
@@ -0,0 +1,156 @@
+preferred OS API.
+
+The code looks like this:
+
+@snippet test/integration/multi-arch/compute.cpp compute
+
+What is going on inside `compute` ?
+  + `load_kernel` opens the correct dynamic library depending on the actual SIMD support detected at
+    runtime via eve::is_supported. Notice the names of the .so files, we will go back to those
+    afterward but understand that they are arbitrary. To do so, we use *Immediately Invoked Lambda
+    Expressions* to perform this initialisation.
+  + Next, we load the `compute_kernel` symbol from the handle we just loaded. As opening and fetching
+    pointer to symbol can be costly, we will make it so those tasks are done once by storing both
+    results in static variables.
+  + Once everything is loaded and setup, we call the function pointer with the appropriate parameters.
+
+Obviously, in realistic settings you would actually care about runtime issues, check that every pointer
+are non-null, and use functions like `dlerror` to find out what would have caused an issue. This is
+left as an exercice for the reader.
+
+## Compiling and Using Dynamic Kernels
+The next part is much more down-to-earth. We need to compile the kernel function inside multiple
+dynamic libraries with different set of options. Then, we need to compile the main binary.
+To simplify this process, **EVE** provides a CMake function named `eve_build_variants` that you can
+access directly when using `find_package` or that can be included manually in your CMake after
+installing the library via `include(eve-multiarch)`.
+
+Its synopsis is:
+```cmake
+eve_build_variants( NAME <name>
+                    TARGET      <target suffix>...
+                    OPTIONS     <architecture specific options>...
+                    [INTERFACE] <name>
+                    [QUIET]
+                    SOURCES  <source files>...
+                  )
+```
+
+`eve_build_variants` adds multiple dynamic libraries targets for compiling the source file specified
+via `SOURCES`. The generated target will be named by combining the `NAME` and each of the `TARGET`
+suffix. For each suffixes, the corresponding options will be used from [OPTIONS].
+
+Optionally, you can pass an `INTERFACE` name to be used for compiling each of the targets.
+Optionally, the `QUIET` flag will remove error messages.
+
+OK, let's say we want to compile our kernel for SSE2, SSE4.1 and AVX2.
+The CMake code will be looking like:
+
+```cmake
+cmake_minimum_required(VERSION 3.19)
+project(eve-multi-arch LANGUAGES CXX)
+
+find_package(eve)
+
+eve_build_variants( NAME        compute
+                    INTERFACE   ${EVE_LIBRARIES}
+                    TARGET      sse2 sse4 haswell
+                    OPTIONS     "-msse2"  "-msse4.1" "-mavx2"
+                    SOURCES     kernel.cpp
+                  )
+
+add_executable(multi-arch main.cpp compute.cpp)
+add_dependencies(multi-arch ${eve_compute_variants})
```

Comment:
Where did `eve_compute_variants` come from? Is that CMAKE ijnect into an extenal scope magic?